src/main/kotlin/io/icure/md/client/filter/Filter.kt

```diff
@@ -74,55 +75,55 @@ fun <T> filter(init: FilterBuilder<T>.() -> Unit): FilterBuilder<T> {
 }
 
 open class FilterBuilder<T>(val parent: FilterBuilder<T>? = null) {
-    var hcp: HealthcareProfessional? = null
-    var proxiedFilterBuilder: ((hcp: HealthcareProfessional?) -> Filter<T>)? = null
+    var dataOwner: DataOwner? = null
+    var proxiedFilterBuilder: ((dataOwner: DataOwner?) -> Filter<T>)? = null
 
-    fun forHcp(hcp: HealthcareProfessional?): FilterBuilder<T> {
-        this.hcp = hcp
+    fun forDataOwner(dataOwner: DataOwner?): FilterBuilder<T> {
```

Comment:
In this method, use the data owner Id in argument, instead of the data owner itself : Otherwise, the user will have to call the getDataOwner service or create a DataOwner dedicated object, whereas we only use its id in filters.